site/content/posts/2024-06-05-githubidentityprovider.md

```diff
@@ -0,0 +1,191 @@
+  client:
+    secretName: my-github-provider-client-secret
+```
+
+## Kubernetes username and group names
+
+The GitHubIdentityProvider resource offers several choices for how your users' Kubernetes usernames and group names should look.
+
+`.spec.claims.username` allows you to choose from:
+- `login`: The user's GitHub username as shown on their profile. Note that a user can change their own username,
+   so this is not recommended for production use with identities from public github.com.
+- `id`: The numeric user ID assigned by GitHub will be used as the username. On public github.com, this can be found for any user
+   by putting their login name into this GitHub API URL: `https://api.github.com/users/cfryanr` (replace `cfryanr` with the login name).
+   This is automatically assigned and immutable for each user.
+- `login:id`: Blends the readability of using login names with the immutability of using IDs by putting both into
+   the Kubernetes usernames, separated by a colon. This keeps your RBAC policies nicely readable.
+
+`spec.claims.groups` allows you to choose from:
+- `name`: GitHub team names can include mixed-case characters, spaces, and punctuation, e.g. `Kube admins!`.
+- `slug`: GitHub slug names are lower-cased, with spaces replaced by hyphens, and other punctuation removed, e.g. `kube-admins`.
+- Either way, the team names will automatically be prefixed by the name of the org in which the team resides, with a `/` separator,
+  e.g. `My-org/kube-admins`. The org name will preserve its case from GitHub.
+
+## Control new and existing sessions by changing org and team memberships on GitHub
+
+Did one of your developers or operators just change teams or leave your enterprise? Fear not. Simply update their
+GitHub organization and/or GitHub team memberships on github.com and Pinniped will respect those changes immediately.
+
+When one of your end users starts a new session, your org and team-based restrictions will apply using your
+updated org and team memberships immediately.
+
+For your end-users with pre-existing ongoing sessions Pinniped will see the new org and team memberships at the next
+session refresh, which happens approximately every 5 minutes in a standard Pinniped configuration for active sessions.
+Your Kubernetes RBAC policies will see the updated group memberships after the next refresh.
+There is no way for your end users to avoid these refreshes without losing access to your clusters.
+
+## Where to read more
+
+This blog post is just a quick overview of this new feature. To learn about how to configure the Pinniped Supervisor
+with this new feature, see:
+
+- The [GitHub configuration guide]({{< ref "docs/howto/supervisor/configure-supervisor-with-github.md" >}}).
+- The [GitHubIdentityProvider resource](https://github.com/vmware-tanzu/pinniped/blob/main/generated/latest/README.adoc#k8s-api-go-pinniped-dev-generated-1-30-apis-supervisor-idp-v1alpha1-githubidentityprovider) documentation.
```

Comment:
Is this anchor permanent? I see `latest` in the path (which seems correct) but `1-30` in the anchor.

Reply:
Updated and force pushed. Good catch.